src/type/value.cpp

```diff
@@ -0,0 +1,130 @@
+#include <type/value.h>
```

Comment:
Was this one of the targets we were gonna consolidate under a src/sql folder? @yangjuns 

That doesn't have to be in this PR, just want to make sure we either document it or have it as part of an issue.